controllers/volumereplicationgroup_controller.go

```diff
@@ -73,10 +74,7 @@ func (r *VolumeReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager) er
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ramendrv1alpha1.VolumeReplicationGroup{}).
 		Watches(&source.Kind{Type: &corev1.PersistentVolumeClaim{}}, pvcMapFun, builder.WithPredicates(pvcPredicate)).
-		// The actual thing that the controller owns is
-		// the VolumeReplication CR. Change the below
-		// line when VolumeReplication CR is ready.
-		Owns(&ramendrv1alpha1.VolumeReplicationGroup{}).
+		Owns(&volrep.VolumeReplication{}).
```

Comment:
Do we need to should watch the VR CR, so that if the user inadvertently changes a VR CR we own, VRG will revert it, aka, set it to what VRG thinks the state should be?  Or, do we get that for free because we own it?

Reply:
We should get that for free, @raghavendrabhat a quick test for this may help ensure things work as expected.

Reply:
Yes. I was able to test it couple of times and it is the behavior I see. i.e. if I change the state of VR CR owned by VRG, then VRG gets a reconcile request and it updates the state of VR CR based on what it thinks is right.

For example, I changed the state of the VR CR from primary to resync and VRG got a reconcile request and again updated the state of VR CR to primary. Checking VR CR showed its state (and also status) is shown as primary.